### PR TITLE
RavenDB-7725

### PR DESCRIPTION
**Pull request by ayende**
- Update to a custom build of ClrMd that allows use to do passive attach.
- On Linux, start the Raven.Debug process, but ask it to wait. Then grant it debug permissions on ourselves, let it run, then remove the permissions.
- No need for any additional security configurations

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Debugging/ThreadsHandler.cs [hunk 1]
```diff
@@ -19,12 +21,6 @@ public class ThreadsHandler : RequestHandler
         [RavenAction("/admin/debug/threads/stack-trace", "GET", AuthorizationStatus.Operator)]
         public Task StackTrace()
         {
-            if (PlatformDetails.RunningOnPosix)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/ThreadsHandler.cs:22]: I think that we need to keep it, but for MacOS?

File: src/Raven.Server/Raven.Server.csproj [hunk 2]
```diff
@@ -102,6 +102,7 @@
     <PackageReference Include="Microsoft.AspNetCore.WebSockets" Version="2.2.1" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.1.0" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.1.0" />
+    <PackageReference Include="Microsoft.Diagnostics.Runtime-ravendb" Version="1.0.7" />
```
Comment (ppekrol) [inline src/Raven.Server/Raven.Server.csproj]: I think we can remove this?

**Review by ppekrol — COMMENTED** [review]
@ayende did you made any changes to clrmd or just compiled master branch?

File: tools/Raven.Debug/CommandLineApp.cs [hunk 3]
```diff
@@ -139,10 +144,20 @@ private static int ExitWithError(string errMsg, CommandLineApplication cmd)
 
             var threadInfoList = new List<ThreadInfo>();
 
-            using (var dataTarget = DataTarget.AttachToProcess(processId, attachTimeout))
+            using (var dataTarget = DataTarget.AttachToProcess(processId, attachTimeout, AttachFlag.Passive))
```
Comment (ppekrol) [inline tools/Raven.Debug/CommandLineApp.cs:147]: Do we want always to attach passively? IIRC it is less accurate, on Windows we could stay with what we had?

**Comment by ayende** [comment]
> @ayende did you made any changes to clrmd or just compiled master branch?

I had to remove a bit of debug code, this is available here: https://github.com/ayende/clrmd

File: src/Raven.Server/Documents/Handlers/Debugging/ThreadsHandler.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Documents/Handlers/Debugging/ThreadsHandler.cs:22]: Yes

File: src/Raven.Server/Raven.Server.csproj (hunk 2 above)
Comment (ayende) [reply src/Raven.Server/Raven.Server.csproj]: Yes

File: tools/Raven.Debug/CommandLineApp.cs (hunk 3 above)
Comment (ayende) [reply tools/Raven.Debug/CommandLineApp.cs:147]: The speed we are seeing in generating stack traces is _huge_, even if it isn't as accurate, that is far more useful

**Review by ppekrol — APPROVED** [review]

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**